src/main/java/wtf/g4s8/examples/spaxos/Proposer.java

```diff
@@ -50,6 +51,8 @@
  * @since 1.0
  */
 public final class Proposer<T> {
+    private final String transactionId;
+    private volatile Boolean crashed;
```

Comment:
@stepanvalyavskiy why not `boolean`? Also, `volatile` on `Boolean` looks strange, since you are working with memory reference to `boolean` holder here, not `boolean` value.

Reply:
I forgot to remove `volatile`, when changed `boolean` to `Boolean`. I need ref type here because I pass it to restart method, which will create new Proposer Instance but with shared `crashed` field. So it will be possible to kill them all together.